### PR TITLE
[Datadict] bug fix on filtering and modifiing unmodified items

### PR DESCRIPTION
**Pull request by xlecours**
- This pull request fix the filtering on modified/unmodified datadictionnary items.
- Also delete non modified items (back to original description)
- Ran phpcs and deleted dead code

### CODE REVIEW

File: modules/datadict/ajax/UpdateDataDict.php [hunk 1]
```diff
@@ -37,32 +37,32 @@
...
 
 if ($user->hasPermission('data_dict_edit')) { //if user has edit permission
-    if ($DB->pselectOne(
-        "SELECT COUNT(*) FROM parameter_type_override " .
-        "WHERE Name =:id",
-        array('id' => $name)
-    )==0) {  //if it doesn't exist
-        // unsafeinsert is needed to allow '<TEST_NAME>_Date_taken' to be inserted
-        $DB->unsafeinsert(
+
```
Comment (xlecours) [inline modules/datadict/ajax/UpdateDataDict.php]: This prevent from inserting in parameter_type_overide when the description is not changed from the original and delete the record when it is back to the original

File: modules/datadict/php/NDB_Menu_Filter_datadict.class.inc [hunk 2]
```diff
@@ -109,74 +109,12 @@ class NDB_Menu_Filter_Datadict extends NDB_Menu_Filter
...
-    /**
-    * Set Data Table Rows
-    *
-    * @param string $count the count of table rows
-    *
-    * @note   Set Data Table Rows
-    * @return bool
-    * @access private
-    */
-    function _setDataTableRows($count)
```
Comment (xlecours) [inline modules/datadict/php/NDB_Menu_Filter_datadict.class.inc]: This is handle in Javascript for AJaxModules

File: modules/datadict/php/NDB_Menu_Filter_datadict.class.inc [hunk 3]
```diff
@@ -109,74 +109,12 @@ class NDB_Menu_Filter_Datadict extends NDB_Menu_Filter
...
-    {
-        // make the SELECT statement
-        $query = "SELECT ";
-        if (is_array($this->columns) && count($this->columns) > 0) {
-            $query .= implode(', ', $this->columns);
-        } else {
-            $query .= "*";
-        }
-        // add the base query
-        if (isset($_REQUEST['Description'])
```
Comment (xlecours) [inline modules/datadict/php/NDB_Menu_Filter_datadict.class.inc]: This is always false

File: modules/datadict/php/NDB_Menu_Filter_datadict.class.inc [hunk 4]
```diff
@@ -202,15 +139,14 @@ class NDB_Menu_Filter_Datadict extends NDB_Menu_Filter
                 if ($val == 'empty') {
                     $query .= " AND COALESCE(pto.description,pt.description) = ''";
                 } elseif ($val=='modified') {
-                    //NO where clause added
+                    $query .= " AND pto.description IS NOT NULL";
```
Comment (xlecours) [inline modules/datadict/php/NDB_Menu_Filter_datadict.class.inc]: This fix the filter for modified (existing overrides)

File: modules/datadict/ajax/UpdateDataDict.php [hunk 5]
```diff
@@ -37,32 +37,32 @@
 } else {
     // Magic quotes is off, so we can just directly use the description
     // since insert() will use a prepared statement.
-    $description = $_REQUEST['description'];
+    $description = trim($_REQUEST['description']);
```
Comment (driusan) [inline modules/datadict/ajax/UpdateDataDict.php]: PHP hasn't supported magic quotes for a fairly long time now (it was removed in PHP 5.4, and the minimum requirement for LORIS is PHP 7.1..)

I think this if statement can just be removed.

**Comment by driusan** [comment]
This is tagged 19.0 but going to the bugfix branch

**Comment by driusan** [comment]
@xlecours This is still tagged 19.0 but going to the bugfix branch and not the release branch. Which one is it supposed to be?

**Comment by zaliqarosli** [comment]
@xlecours looks good, everything's working as should. 

side note: it would be a good idea to add to the data dictionary help file that the description column can be directly edited on the table itself as its not obvious. @driusan not sure what the plan is for updating help files

**Comment by driusan** [comment]
@zaliqarosli You can send a separate PR to add more detail to the help file if you think it can be improved. You should just need to edit the markdown file in the help/ directory for the appropriate page.

File: modules/datadict/php/datadict.class.inc [hunk 6]
```diff
@@ -199,15 +138,14 @@ class Datadict extends \NDB_Menu_Filter
                 if ($val == 'empty') {
                     $query .= " AND COALESCE(pto.description,pt.description) = ''";
                 } elseif ($val=='modified') {
-                    //NO where clause added
+                    $query .= " AND pto.description IS NOT NULL";
```
Comment (ridz1208) [inline modules/datadict/php/datadict.class.inc]: @xlecours isn't it possible that a field IS modified and the description IS null or is that not supposed to be done ?

i.e. a parameter description is "blablabla" and is was overridden with NULL ??

Comment (zaliqarosli) [reply modules/datadict/php/datadict.class.inc]: when testing, if the description is null, it isnt considered modified, it's considered empty, but the empty filter isn't working as should (i assumed not part of this PR)

File: modules/datadict/php/datadict.class.inc [hunk 7]
```diff
@@ -108,72 +108,12 @@ class Datadict extends \NDB_Menu_Filter
...
-                $this->tpl_data['items'][$x][$i]['value'] = $val;
-
-                $i++;
-            }
-
-            $x++;
-        }
         return true;
     }
-    /**
```
Comment (ridz1208) [inline modules/datadict/php/datadict.class.inc]: @xlecours not sure why this is considered dead ? the function is being used and it is different then the parent function

File: modules/datadict/php/datadict.class.inc (hunk 6 above)
Comment (zaliqarosli) [reply modules/datadict/php/datadict.class.inc]: the null value also gets overridden by the original description on the front-end but the change is logged in parameter_type_override table

**Comment by zaliqarosli** [comment]
@xlecours sorry, i missed something! on 19.0, an empty description field gets saved as empty whereas, here, the value resets to the original description

File: modules/datadict/php/datadict.class.inc (hunk 7 above)
Comment (ridz1208) [reply modules/datadict/php/datadict.class.inc]: okai after further tracking I realised that the additional if statement is no longer being used. so this is fine

**Pull request merged**